### PR TITLE
add first blueprint

### PR DESCRIPTION
**Pull request by gtema**

### CODE REVIEW

File: doc/source/cce_vault.rst [hunk 1]
```diff
@@ -0,0 +1,618 @@
...
+       bound_service_account_names=zookeeper \
+       bound_service_account_namespaces=zookeeper \
+       policies=tls-zk-ro \
+       ttl=2h
+
+With this done token of the service account zookeeper in the zookeeper
+namespace is able to access to the vault for reading secrets located under
+`/secret/tls` path. And since it is higly recommended to follow the least
+required privilege principle only read only access to the TLS data is granted.
+A time to live of three hours is being used here meaning that once application
```
Comment (kucerakk) [inline doc/source/cce_vault.rst]: "time to live of three hours...used during next two hours"

File: doc/source/cce_vault.rst [hunk 2]
```diff
@@ -0,0 +1,618 @@
+===============================================
+Secrets management with CCE and Hashicorp Vault
+===============================================
+
+Most modern IT setups are composed of several subsystems like databases, object
+stores, master controller, node access, and more. To access one component from
+another, some form of credentials are required. Configuring and storing these
+secrets directly in the components is considered an antipattern, since a
```
Comment (vladimirhasko) [inline doc/source/cce_vault.rst]: an antipattern or as an antipattern?

File: doc/source/cce_vault.rst [hunk 3]
```diff
@@ -0,0 +1,618 @@
...
+vulnerability of one component may iteratively affect the security of the whole
+setup.
+
+With centralized secret management it becomes unnecessary to keep secrets used
+by various application spreaded across DevOps environments. This helps to close
```
Comment (vladimirhasko) [inline doc/source/cce_vault.rst]: applications

File: doc/source/cce_vault.rst [hunk 4]
```diff
@@ -0,0 +1,618 @@
...
+some security attack vectors (like `secret sprawl
+<https://www.hashicorp.com/resources/what-is-secret-sprawl-why-is-it-harmful>`_,
+`security islands <https://www.conjur.org/blog/security-islands/>`_), but
+usually introduces a problem of the so-called `Secret Zero
+<https://www.hashicorp.com/resources/secret-zero-mitigating-the-risk-of-secret-introduction-with-vault>`_
+as a key to the key storage.
+
+Vault is an open-source software, provided and maintained by Hashicorp, that
+addresses this very problem. It is considered one of the reference solutions
```
Comment (vladimirhasko) [inline doc/source/cce_vault.rst:22]: this very problem is quite rare phrase. Sounds weird to me but if you are fine with that feel free to keep it

File: doc/source/cce_vault.rst (hunk 1 above)
Comment (vladimirhasko) [inline doc/source/cce_vault.rst]: again three hours vs two hours

File: doc/source/cce_vault.rst (hunk 4 above)
Comment (gtema) [reply doc/source/cce_vault.rst:22]: Sentence from Nils ;-)

**Comment by gtema** [comment]
btw, instead of many times having .yaml + .shell I can (at least in some cases) switch to `kubectl apply -f - <<-EOF`

**Comment by vladimirhasko** [comment]
> btw, instead of many times having .yaml + .shell I can (at least in some cases) switch to `kubectl apply -f - <<-EOF`

sounds good to me
just the first approach offers more re-usability as those snippets yaml you can reuse in other examples if required.

**Comment by gtema** [comment]
so, do we maybe want to merge it and improve further with PRs or do we wait to reach better state?

**Review by kucerakk — APPROVED** [review]

**Review by vladimirhasko — APPROVED** [review]

**Comment by vladimirhasko** [comment]
> her with PRs or do we wait t

with further PRs, let's shoot it out now

**Pull request merged**